pkg/cloud/cloud.go

```diff
@@ -18,6 +19,38 @@ import (
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/vsphere"
 )
 
+// cloudConfigTransformer function transforms the source config map using the input infrastructure.config.openshift.io object.
+// only the data and binaryData field of the output ConfigMap will be respected by consumer of the transformer.
+type cloudConfigTransformer func(source *corev1.ConfigMap, infra *configv1.Infrastructure) (*corev1.ConfigMap, error)
```

Comment:
Nit, why do we need to return a pointer to the CM, surely the `source` input and configmap return values are the same pointer, or have I misunderstood?